src/MagnumPlugins/DdsImporter/DdsImporter.cpp

```diff
@@ -0,0 +1,356 @@
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include <algorithm>
+#include <cstring>
+#include <fstream>
+#include <sstream>
+#include <Corrade/Containers/ArrayView.h>
+#include <Corrade/Utility/Debug.h>
+#include <Magnum/ColorFormat.h>
+#include <Magnum/Trade/ImageData.h>
+#include <MagnumPlugins/DdsImporter/DdsImporter.h>
+#include <Magnum/Math/Vector3.h>
+#include <Magnum/Math/Vector4.h>
+
+#ifdef MAGNUM_TARGET_GLES2
+#include <Magnum/Context.h>
+#include <Magnum/Extensions.h>
+#endif
+
+namespace Magnum { namespace Trade {
+
+/**
+ * Flags to indicate which members of a DdsHeader contain valid data.
+ */
+enum DdsDescriptionFlag: UnsignedInt {
+    Caps = 0x00000001,       //!< Caps
+    Height = 0x00000002,     //!< Height
+    Width = 0x00000004,      //!< Width
+    Pitch = 0x00000008,      //!< Pitch
+    PixelFormat = 0x00001000,//!< PixelFormat
+    MipMapCount = 0x00020000,//!< MipMapCount
+    LinearSize = 0x00080000, //!< LinearSize
+    Depth = 0x00800000       //!< Depth
+};
+
+/**
+ * Direct Draw Surface pixel format.
+ */
+enum DdsPixelFormat: UnsignedInt {
+    AlphaPixels = 0x00000001,//!< AlphaPixels
+    FourCC = 0x00000004,     //!< FourCC
+    RGB = 0x00000040,        //!< RGB
+    RGBA = 0x00000041        //!< RGBA
+};
+
+/**
+ * Specifies the complexity of the surfaces stored.
+ */
+enum DdsCaps1: UnsignedInt {
+    /**
+     * Set for files that contain more than one surface (a mipmap, a cubic environment map,
+     * or mipmapped volume texture).
+     */
+    Complex = 0x00000008,
+    Texture = 0x00001000,//!< Texture (required).
+    MipMap = 0x00400000  //!< Is set for mipmaps.
+};
+
+/**
+ * Additional detail about the surfaces stored.
+ */
+enum DdsCaps2: UnsignedInt {
+    Cubemap = 0x00000200,         //!< Cubemap
+    CubemapPositiveX = 0x00000400,//!< CubemapPositiveX
+    CubemapNegativeX = 0x00000800,//!< CubemapNegativeX
+    CubemapPositiveY = 0x00001000,//!< CubemapPositiveY
+    CubemapNegativeY = 0x00002000,//!< CubemapNegativeY
+    CubemapPositiveZ = 0x00004000,//!< CubemapPositiveZ
+    CubemapNegativeZ = 0x00008000,//!< CubemapNegativeZ
+    CubemapAllFaces = 0x0000FC00, //!< CubemapAllFaces
+    Volume = 0x00200000           //!< Volume
+};
+
+/**
+ * Compressed texture types.
+ */
+enum DdsCompressionTypes: UnsignedInt {
+    DXT1 = 0x31545844, //!< (MAKEFOURCC('D','X','T','1')).
+    DXT2 = 0x32545844, //!< (MAKEFOURCC('D','X','T','2')), not supported.
+    DXT3 = 0x33545844, //!< (MAKEFOURCC('D','X','T','3')).
+    DXT4 = 0x34545844, //!< (MAKEFOURCC('D','X','T','4')), not supported.
+    DXT5 = 0x35545844, //!< (MAKEFOURCC('D','X','T','5')).
+    DXT10 = 0x30315844 //!< (MAKEFOURCC('D','X','1','0')), not supported.
+};
+
+inline UnsignedInt clampSize(UnsignedInt size) {
+    return (size <= 0) ? 1 : size = 1;
```

Comment:
Yeah, I tried that, but my compiler complained and I didn't have the patience to deal with it yet :) I will try again later.